src/components/cylc/graph/Graph.vue

```diff
@@ -0,0 +1,631 @@
+      if (this.tippy !== null) {
+        this.tippy.hide()
+      }
+      this.cytoscapeInstance
+        .elements()
+        .remove()
+      this.cytoscapeInstance
+        .style(cytoscapeDefaultStyle)
+        .update()
+      this.cytoscapeInstance.add(this.graphData)
+      this.setupUndoRedo(this.cytoscapeInstance)
+      const loaded = await this.cytoscapeInstance
+        .elements()
+        .layout(this.layoutOptions)
+        .run()
+      if (!loaded) {
+        throw new Error('There was an error loading the graph view!')
+      }
+      this.loading = false
+    },
+
+    /**
+     * Called by the UI when the user switches a layout using some UI component (button/dropdown/etc). Its
+     * responsibility is simply to change the attributes related to the layout, and then fi`re an updateLayout call.
+     * @param {string} layoutName
+     */
+    switchLayout (layoutName) {
+      this.layoutName = layoutName
+      this.layoutReady = true
+      this.layoutStopped = false
+      this.updateLayout()
+    },
+
+    /**
+     * Called after the user switches layout. Its responsibility is to change the this.layoutOptions (dagre by default)
+     * to the correct options, and then call this.doLayout(newOptions).
+     *
+     * We set the this.layoutOptions as in the this.updateGraph it is used to apply the layout to the newly
+     * created elements.
+     */
+    updateLayout () {
+      switch (this.layoutName) {
+        case 'dagre':
+          this.layoutOptions = dagreOptions
+          this.runLayout(this.cytoscapeInstance, dagreOptions)
+          break
+        case 'hierarchical':
+          this.cytoscapeInstance
+            .elements()
+            .hca({
+              mode: 'threshold',
+              threshold: 25,
+              distance: 'euclidean', // euclidean, squaredEuclidean, manhattan, max
+              preference: 'mean', // median, mean, min, max,
+              damping: 0.8, // [0.5 - 1]
+              minIterations: 100, // [optional] The minimum number of iterations the algorithm will run before stopping (default 100).
+              maxIterations: 1000, // [optional] The maximum number of iterations the algorithm will run before stopping (default 1000).
+              attributes: [
+                (node) => {
+                  return node.data('weight')
+                }
+              ]
+            })
+          this.layoutOptions = hierarchicalOptions
+          this.runLayout(this.cytoscapeInstance, hierarchicalOptions)
+          break
+        default:
+          // Should never happen!
+          this.layoutOptions = dagreOptions
+          this.runLayout(this.cytoscapeInstance, dagreOptions)
+          break
+      }
+    }
+  }
+}
+</script>
+
+<style lang="scss">
+@import '~@/styles/cytoscape/html-label.scss';
+@import '~@/styles/cytoscape/cytoscape-custom.scss';
+@import '~@/styles/cytoscape/panzoom.css';
+@import '~tippy.js/themes/light-border.css';
+</style>
```

Comment:
There were 2 blocks of `css` and `scss`, but `scss` can handle `css` too. The Toppy CSS was imported in the JS part (webpack is able to handle that, but easier to have that here I guess)